### PR TITLE
Add com.github.IsmaelMartinez.teams_for_linux

### PR DESCRIPTION
**Pull request by gabops**
From its official repo https://github.com/IsmaelMartinez/teams-for-linux:
> Unofficial Microsoft Teams client for Linux using Electron. It uses the Web App and wraps it as a standalone application using Electron.

I have been testing it locally with very good results. The sharing screen, audio and video in meetings and normal usage work like a charm.

### CODE REVIEW

**Comment by LeandroStanger** [comment]
bot, build com.GitHub.ismael martinez.teams_for_linux

**Comment by gabops** [comment]
bot, com.github.IsmaelMartinez.teams_for_linux

**Comment by gabops** [comment]
bot, build com.github.IsmaelMartinez.teams_for_linux

**Comment by gabops** [comment]
bot, build com.github.IsmaelMartinez.teams_for_linux

**Comment by nedrichards** [comment]
Mentioned upstream at https://github.com/IsmaelMartinez/teams-for-linux/issues/41

**Comment by nedrichards** [comment]
@gabops is there a reason you haven't built this from source using something like https://github.com/flatpak/flatpak-builder-tools/tree/master/node ?

**Comment by gabops** [comment]
> @gabops is there a reason you haven't built this from source using something like https://github.com/flatpak/flatpak-builder-tools/tree/master/node ?

At the beginning I was compiling nodejs and trying to build the project from source by using flatpak-builder-tools/node you mentioned, but after a long time having several frustrating issues with some electron js dependencies I had a look to other projects and I took as example the "io.atom.Atom" manifest. That thing of using the deb package is clearly not the most right nor elegant but the simpler.

If this is clearly not the way I can continue researching and trying to make it work from source.

**Comment by nedrichards** [comment]
@gabops if you've tried and there's actual blockers, then a repackage is OK, although feel free to share and we'll see if we can help - although it's harder to make sure you distribute the source code as well (as I remember you need to add that as an archive and it should all get worked out)

**Comment by gabops** [comment]
> @gabops if you've tried and there's actual blockers, then a repackage is OK, although feel free to share and we'll see if we can help - although it's harder to make sure you distribute the source code as well (as I remember you need to add that as an archive and it should all get worked out)

@nedrichards Thanks for that. I have been working today in the build from source. I found how to fix the issues I was having with the requirements. Is looking good. Tomorrow I will push some changes. Could you please add a WIP label to be sure this is not being merged as it is right now?

**Comment by nedrichards** [comment]
Sure, although it won't be currently as the build is failing right now ;-)

**Comment by gabops** [comment]
bot, build com.github.IsmaelMartinez.teams_for_linux

**Comment by gabops** [comment]
bot, build com.github.IsmaelMartinez.teams_for_linux

File: com.github.IsmaelMartinez.teams_for_linux.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  <id>com.github.IsmaelMartinez.teams_for_linux</id>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-3.0</project_license>
+  <name>teams-for-linux</name>
+  <summary>Unofficial Microsoft Teams client for Linux using Electron</summary>
+  <description>
+    <p>Unofficial Microsoft Teams client for Linux using Electron.</p> 
+    <p>It uses the Web App and wraps it as a standalone application using Electron.</p>
+  </description>
+  <categories>
```
Comment (bilelmoussaoui) [inline com.github.IsmaelMartinez.teams_for_linux.appdata.xml]: The categories should be put on the desktop file. They are merged with the appdata file later.

File: com.github.IsmaelMartinez.teams_for_linux.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    <category>Application</category>
+    <category>Network</category>
+  </categories>
+  <url type="homepage">https://github.com/IsmaelMartinez/teams-for-linux</url>
+  <url type="help">https://github.com/IsmaelMartinez/teams-for-linux/blob/develop/KNOWN_ISSUES.md</url>
+  <url type="bugtracker">https://github.com/IsmaelMartinez/teams-for-linux/issues</url>
+  <launchable type="desktop-id">teams-for-linux.desktop</launchable>
```
Comment (bilelmoussaoui) [inline com.github.IsmaelMartinez.teams_for_linux.appdata.xml]:
```suggestion
  <launchable type="desktop-id">com.github.IsmaelMartinez.teams_for_linux.desktop</launchable>
```

File: com.github.IsmaelMartinez.teams_for_linux.desktop [hunk 3]
```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Name=Microsoft Teams for Linux
+Comment=Unofficial client for Microsoft Teams for Linux
+Exec=teams-for-linux %U
+Terminal=false
+Type=Application
+Icon=teams-for-linux.png
+StartupWMClass=teams-for-linux
+MimeType=application/sip;
+Categories=Chat;Network;Office;
```
Comment (bilelmoussaoui) [inline com.github.IsmaelMartinez.teams_for_linux.desktop]: You could use some keywords here.

File: com.github.IsmaelMartinez.teams_for_linux.desktop [hunk 4]
```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Name=Microsoft Teams for Linux
+Comment=Unofficial client for Microsoft Teams for Linux
+Exec=teams-for-linux %U
+Terminal=false
+Type=Application
+Icon=teams-for-linux.png
```
Comment (bilelmoussaoui) [inline com.github.IsmaelMartinez.teams_for_linux.desktop]:
```suggestion
Icon=com.github.IsmaelMartinez.teams_for_linux
```
The icon name, desktop file, appdata file should have the same name as your application id.

File: com.github.IsmaelMartinez.teams_for_linux.yml [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+base: org.electronjs.Electron2.BaseApp
+base-version: '18.08'
+runtime: org.freedesktop.Platform
+runtime-version: '18.08'
+sdk: org.freedesktop.Sdk
+command: teams-for-linux
+separate-locales: false
+rename-desktop-file: teams-for-linux.desktop
+rename-appdata-file: teams-for-linux.appdata.xml
+rename-icon: teams-for-linux
```
Comment (bilelmoussaoui) [inline com.github.IsmaelMartinez.teams_for_linux.yml]: Not needed once you rename the icon on the appdata file.

File: com.github.IsmaelMartinez.teams_for_linux.yml [hunk 6]
```diff
@@ -0,0 +1,89 @@
+app-id: com.github.IsmaelMartinez.teams_for_linux
+base: org.electronjs.Electron2.BaseApp
+base-version: '18.08'
+runtime: org.freedesktop.Platform
+runtime-version: '18.08'
+sdk: org.freedesktop.Sdk
+command: teams-for-linux
+separate-locales: false
+rename-desktop-file: teams-for-linux.desktop
```
Comment (bilelmoussaoui) [inline com.github.IsmaelMartinez.teams_for_linux.yml]: Just install the appdata/desktop files with the right name.

File: com.github.IsmaelMartinez.teams_for_linux.yml [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+  - --filesystem=xdg-run/dconf
+  - --filesystem=~/.config/dconf:ro
+  - --env=DCONF_USER_CONFIG_DIR=.config/dconf
+  - --talk-name=ca.desrt.dconf
+  - --talk-name=org.freedesktop.Notifications
+  - --talk-name=org.kde.StatusNotifierWatcher
+  - --env=XDG_CURRENT_DESKTOP=Unity
+  - --talk-name=org.gtk.Notifications
+  - --talk-name=org.kde.StatusNotifierWatcher
+build-options:
```
Comment (bilelmoussaoui) [inline com.github.IsmaelMartinez.teams_for_linux.yml]: Those are the default build-options already, you can remove them.

File: com.github.IsmaelMartinez.teams_for_linux.yml [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+    build-commands:
+      - yarn --offline --ignore-scripts install
+      - yarn --offline --ignore-scripts run dist:linux dir -c.npmRebuild=false
+      - cp -r dist/linux-unpacked /app/teams-for-linux
+      - ln -s /app/teams-for-linux/teams-for-linux /app/bin/teams-for-linux
+      - | 
+          for icon in $(ls --ignore '1024*' -v1 build/icons/)
+          do
+            mkdir -p /app/share/icons/hicolor/$(basename -s '.png' $icon)/apps/
+            install -Dm644 build/icons/$icon /app/share/icons/hicolor/$(basename -s '.png' $icon)/apps/teams-for-linux.png
```
Comment (bilelmoussaoui) [inline com.github.IsmaelMartinez.teams_for_linux.yml]:
```suggestion
            install -Dm644 build/icons/$icon /app/share/icons/hicolor/$(basename -s '.png' $icon)/apps/com.github.IsmaelMartinez.teams_for_linux.png
```

File: com.github.IsmaelMartinez.teams_for_linux.yml [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+          done
+      - install -Dm644 teams-for-linux.appdata.xml -t /app/share/metainfo
```
Comment (bilelmoussaoui) [inline com.github.IsmaelMartinez.teams_for_linux.yml]:
```suggestion
      - install -Dm644 com.github.IsmaelMartinez.teams_for_linux.appdata.xml -t /app/share/metainfo
```

File: com.github.IsmaelMartinez.teams_for_linux.yml [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+      - install -Dm644 teams-for-linux.desktop -t /app/share/applications
```
Comment (bilelmoussaoui) [inline com.github.IsmaelMartinez.teams_for_linux.yml]:
```suggestion
      - install -Dm644 com.github.IsmaelMartinez.teams_for_linux.desktop -t /app/share/applications
```

File: com.github.IsmaelMartinez.teams_for_linux.yml [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+        electron_config_cache: /run/build/teams-for-linux/electron-cache/
+        PATH: /app/bin:/run/build/teams-for-linux/node_modules/.bin:/bin
+    sources:
+      - generated-sources.json
+      - type: git
+        url: https://github.com/IsmaelMartinez/teams-for-linux.git
+        tag: v0.3.0
+      - type: file
+        path: com.github.IsmaelMartinez.teams_for_linux.appdata.xml
+        dest-filename: teams-for-linux.appdata.xml
```
Comment (bilelmoussaoui) [inline com.github.IsmaelMartinez.teams_for_linux.yml]:
```suggestion
```

File: com.github.IsmaelMartinez.teams_for_linux.yml [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+      - type: file
+        path: com.github.IsmaelMartinez.teams_for_linux.desktop
+        dest-filename: teams-for-linux.desktop
```
Comment (bilelmoussaoui) [inline com.github.IsmaelMartinez.teams_for_linux.yml]:
```suggestion
```

**Comment by gabops** [comment]
bot, build com.github.IsmaelMartinez.teams_for_linux

**Comment by nedrichards** [comment]
this looks good to me.

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/com.github.IsmaelMartinez.teams_for_linux

**Pull request closed without merging**